sys/console/full/src/uart_console.c

```diff
@@ -230,19 +230,16 @@ uart_console_rx_char_event(struct os_event *ev)
     static int b = -1;
     int sr;
     int ret;
+    bool full;
+
+    full = uart_console_ring_is_full(&cr_rx);
 
     /* We may have unhandled character - try it first */
     if (b >= 0) {
         ret = console_handle_char(b);
         if (ret < 0) {
-            return;
+            goto done;
```

Comment:
I think in this case it does not make sense to restart RX since we did not pull any character from ring buffer so RX will be stalled again.

Reply:
Good point.